Upgrade dbplyr generics

florianm opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
This PR upgrades dbplyr generics from v1 to v2 to prevent breakage once the v1 generics are removed from dbplyr.

## Description
<!--- Describe your changes in detail -->

- [x] Follow the upgrade guide at https://dbplyr.tidyverse.org/articles/backend-2.html#nd-edition
- [x] DESCRIPTION: Bump version (suggest 0.7.1 once merged), bump min versions for jsonlite, dplyr, dbplyr to latest available, update author email for FM, bump RoxygenNote
- [x] ckanr-package.R: Upgrade syntax to fix Roxygen warning, add new authors from DESCRIPTION
- [x] Add dbplyr_edition as per upgrade guide (new man page)
- [x] Upgrade dplyr.R generics and imports as per upgrade guide
- [x] Document package to rebuild NAMESPACE
- [ ] TODO NEWS

## Related Issue
<!--- if this closes an issue make sure include e.g., "fix #4"
or similar - or if just relates to an issue make sure to mention
it like "#4" -->

Fix #187 

## Example
<!--- if introducing a new feature or changing behavior of existing
methods/functions, include an example if possible to do in brief form -->

<!--- Did you remember to include tests? Unless you're just changing
grammar, please include new tests for your change -->
This PR upgrades internal syntax (dbplyr generics). It replaces dplyr generics with differently named dbplyr generics - neither are used explicitly elsewhere, so I assume this change will not break existing workflows. It will however require `ckanr` users to also upgrade their use of exposed dplyr generics to the newer dbplyr equivalents. A well-written release note in NEWS will help (coming in this PR).

I did not have a working CKAN test instance at hand when writing this PR. Happy to test and fix if needed if I could get access to a CKAN environment.

fjuniorr commented:
> How do you feel about pinning and upgrading all the dependencies?

My overall preference would be to not pin the dependencies to allow the installation on older environments.

> The package docs mention that `ds_create_dataset` will be removed in favour of `resource_create`. This PR suggests a new package version (0.8.0). Would you like me to drop `ds_create_dataset` in this PR or leave it up to you?

Would you mind opening a separate PR for this? I'm still not familiar with the codebase and it will make it easier for me to digest piecemeal changes.

> @fjuniorr while I'm getting a local CKAN up and running, have you got any pointer why the GH action workflow can't find the CKAN test instance?

The workflow needed access to the `CKANR_TEST_URL` and `CKANR_TEST_BEHAVIOUR` env vars but they were defined as secrets, so they were not passed to the runner when a workflow is triggered from a forked repository.

I'm fixing this on https://github.com/ropensci/ckanr/pull/212 and managed to run the tests on [ubuntu](https://github.com/ropensci/ckanr/actions/runs/9262669643) but some other issue came up that still needs investigating related to the `CKANR_TEST_BEHAVIOUR` on windows and macos runners.

florianm commented:
@fjuniorr thanks for the info!

re dependencies:
This PR only needs to pin dbplyr and dplyr to support the new dbplyr backend API.
I've moved the idea of requiring R >= 4.1 to #213.

re deprecation: moved to #214.

Let me try out an idea re #212.

As per discussion at #187, this change could run deeper than just to upgrade the geenrics as I did - we might want to drop most of dplyr.R entirely and use a vanilla DBI connection instead.
